tower-http/src/services/fs/serve_dir.rs

```diff
@@ -183,7 +198,17 @@ impl<ReqBody> Service<Request<ReqBody>> for ServeDir {
                 });
 
             let file = File::open(full_path).await?;
-            Ok(Output::File(file, mime, buf_chunk_size))
+            let cache_control = if let Some(max_age) = max_age {
+                Some(HeaderValue::from_str(&format!("max-age={}", max_age.as_secs())).unwrap())
```

Comment:
Like I wrote [here](https://github.com/tower-rs/tower-http/pull/163#discussion_r749189630)

> To avoid allocating a string for each response (only to create a `HeaderValue`) I think should instead change the field from a `Duration` (or rather `Option<Duration>`) to `Option<HeaderValue>` and then generate the header value directly when the user sets it. Cloning a `HeaderValue` is cheaper since its reference counted.

So this needs to happen in the `max_age` builder method, not here.